File: vortexdbg-api/src/main/java/com/vortexdbg/arm/context/RegisterContext.java
package com.vortexdbg.arm.context;

import com.vortexdbg.pointer.VortexdbgPointer;

public interface RegisterContext {

    /**
     * @param index 0 based
     */
    int getIntArg(int index);

    /**
     * @param index 0 based
     */
    long getLongArg(int index);

    /**
     * @param index 0 based
     */
    VortexdbgPointer getPointerArg(int index);

    long getLR();

    VortexdbgPointer getLRPointer();

    VortexdbgPointer getPCPointer();

    /**
     * sp
     */
    VortexdbgPointer getStackPointer();

    int getIntByReg(int regId);
    long getLongByReg(int regId);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/arm/context/RegisterContext.kt
package com.vortexdbg.arm.context

import com.vortexdbg.pointer.VortexdbgPointer

interface RegisterContext {

    /**
     * @param index 0 based
     */
    fun getIntArg(index: Int): Int

    /**
     * @param index 0 based
     */
    fun getLongArg(index: Int): Long

    /**
     * @param index 0 based
     */
    fun getPointerArg(index: Int): VortexdbgPointer?

    fun getLR(): Long

    fun getLRPointer(): VortexdbgPointer

    fun getPCPointer(): VortexdbgPointer

    /**
     * sp
     */
    fun getStackPointer(): VortexdbgPointer

    fun getIntByReg(regId: Int): Int
    fun getLongByReg(regId: Int): Long

}
